### ISSUE TITLE
Support WebSockets

### ISSUE THREAD

**Issue by LorbusChris** [pocesar/node-stratum#20]
Hello,
this is just an idea, as HTTP and TCP are both already supported in this project.
Afaik a WebSocket connection starts out as HTTP, a handshake occurs and the protocol is changed to TCP. One could probably you a library such as ws for that.

**Comment by pocesar** [comment]
yeah, it could. in the refactor I'm doing, I'm trying to decouple most of hard coded functionality so it could actually facilitate using other protocols, like HTTP, RPC2 itself, gRPC, etc, because right now, doing a stratum proxy isn't very straightforward and needs a lot of jumping through hoops to get there
